### DISCUSSION TITLE
Erasure Coding disposition

### DISCUSSION THREAD

**Discussion by zhl19911203** [minio/minio#20312] [Q&A] [answered]
There are a few questions to consult
1、N (ERASURE SET SIZE) = K (DATA) + M (PARITY)
When storing data, how do you know whether a disk is storing data blocks or parity blocks?

2、When a disk fails, can you distinguish between a data block and a parity block? For example, if a total of 16 disks are used, EC:8 is used, and 8 disks are damaged, but all 8 disks are stored in parity blocks，Can the files be recovered in this way?
3、xl.meta and part.1, the rules for splitting, such as 10M or other sizes
4、xl.meta and part.1 which is the data block and which is the parity block? Is there a correspondence between this?

**Comment by marktheunissen** [comment, marked as answer]
Hi @zhl19911203 

1. That is determined by our internal algorithm, you can read the source code if you really want to know exactly how it happens. But this is not something that the user should be concerned about.
2. MinIO is tolerant of the failure of both parity and data drives. It does not matter whether a parity or data block fails, because the data can be regenerated from either kind.
3. This is handled internally, once again please read the source code if you want to know, or read about Reed-Solomon EC. We have optimised this for performance. 
4. No that's not how it works, those are shards of one type, not multiple types

You should read up about Erasure Encoding generally:
- https://min.io/docs/minio/linux/operations/concepts/erasure-coding.html

And external links:
- https://www.youtube.com/watch?v=jgO09opx56o
- https://www.backblaze.com/blog/reed-solomon/

**Reply by zhl19911203** [reply]
Thanks for the answer
